x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -469,13 +470,26 @@ export const fillMissingCustomFields = ({
   const customFieldsKeys = new Set(customFields.map((customField) => customField.key));
   const missingCustomFields: CaseRequestCustomFields = [];
 
+  // only populate with the default value required custom fields missing from the request
   for (const confCustomField of customFieldsConfiguration) {
     if (!customFieldsKeys.has(confCustomField.key)) {
-      missingCustomFields.push({
-        key: confCustomField.key,
-        type: confCustomField.type,
-        value: null,
-      });
+      if (
+        confCustomField.required &&
+        confCustomField?.defaultValue !== null &&
+        confCustomField?.defaultValue !== undefined
+      ) {
+        missingCustomFields.push({
```

Comment:
If in the future we add new attributes in the `CaseCustomField` we will not get a TS error. What about spreading on top?

```
{
        ...customField, <--- spreading
        key: customField.key,
        type: customField.type,
        value: customFieldConfiguration.defaultValue,
      } as CaseCustomField;
```

Reply:
But there is no `customField` here, we are filling the **missing** ones :D 

Additionally, I cannot spread the configuration because they have different properties.

Reply:
You are right, I missed that. Still, we need a way to get notified if we add more properties to the `CaseCustomField`, right? Will the following work?

```
{
        key: customField.key,
        type: customField.type,
        value: customFieldConfiguration.defaultValue as CaseCustomField['value'],
      }
```

Reply:
Talked offline, it didn't work :(